src/wallet/rpcwallet.cpp

```diff
@@ -1767,6 +1768,7 @@ UniValue gettransaction(const UniValue& params, bool fHelp)
     CAmount nNet = nCredit - nDebit;
     CAmount nFee = (wtx.IsFromMe(filter) ? wtx.GetValueOut() - nDebit : 0);
 
+    entry.pushKV("version", wtx.nVersion);
```

Comment:
@str4d I think that we should move toward not requiring multiple calls to get all the information about a single transaction; at most, we should perhaps have an additional argument to the RPC call that specifies the level of detail. But the version seems like something fundamental that we should just return.

Reply:
I'm weakly against this because I think we should have a higher-level look at what the RPCs contain, but if you're okay with us potentially removing this field again in a future release, then I'm not blocking.